Failing test: test_piece_creation_edge_cases

Some of these test cases are not raising a ValueError:

def test_piece_creation_edge_cases():
    # Test edge cases for piece creation
    with pytest.raises(ValueError):
        Piece(None)
    with pytest.raises(ValueError):
        Piece("InvalidType")

Stacktrace:


============================= test session starts ==============================
platform linux -- Python 3.13.1, pytest-8.3.4, pluggy-1.5.0
rootdir: /home/runner/work/copilot-tdd-exercise/copilot-tdd-exercise
collected 10 items
tests/test_board.py .....                                                [ 50%]
tests/test_game.py .                                                     [ 60%]
tests/test_pieces.py ..F.                                                [100%]
=================================== FAILURES ===================================
________________________ test_piece_creation_edge_cases ________________________
    def test_piece_creation_edge_cases():
        # Test edge cases for piece creation
>       with pytest.raises(ValueError):
E       Failed: DID NOT RAISE <class 'ValueError'>
tests/test_pieces.py:[26](https://github.com/Orodruin-org/copilot-tdd-exercise/actions/runs/12909332021/job/35996760972?pr=3#step:5:27): Failed
=========================== short test summary info ============================
FAILED tests/test_pieces.py::test_piece_creation_edge_cases - Failed: DID NOT RAISE <class 'ValueError'>

--- a/tetris/pieces.py
+++ b/tetris/pieces.py
@@ -11,6 +11,8 @@ class TetrominoType(Enum):
 
 class Piece:
     def __init__(self, tetromino_type):
+        if tetromino_type is None or not isinstance(tetromino_type, TetrominoType):
+            raise ValueError("Invalid tetromino type")
         self.type = tetromino_type
         # For demonstration, store a shape as a list of coordinates or something similar
         # e.g., for an I piece, a simple placeholder could be:

